src/bin/vhost_user_fs.rs

```diff
@@ -404,6 +397,24 @@ fn main() {
         process::exit(1);
     }
 
+    loop {
+        if fs_backend.read().unwrap().mem.is_some() && fs_backend.read().unwrap().vrings.is_some() {
+            if daemon.acked_protocol_features() & VhostUserProtocolFeatures::SLAVE_REQ.bits() != 0 {
+                loop {
+                    if fs_backend.read().unwrap().vu_req.is_some() {
+                        break;
+                    }
+                }
+                if let Err(e) = fs_backend.write().unwrap().run() {
+                    error!("Failed to run worker thread for vhost-user-fs: {:?}", e);
+                }
+            } else if let Err(e) = fs_backend.write().unwrap().run() {
+                error!("Failed to run worker thread for vhost-user-fs: {:?}", e);
+            }
+            break;
+        }
+    }
+
```

Comment:
Oh, gosh, please no. Worker threads must support updating their properties, when necessary. This is just cheating in an incredible ugly way.

Reply:
Yes, I understand it. It's not a good way to handle the thing. I just take a look in vm-virtio/src/vhost_user/fs, and I move the set_slave_req_fd ahead of other vhost_user protocol handling, this could ensure both vu_req and mem are not none when we start worker thread for fs backend. So never need to loop here, just remove this part.